### PR TITLE
feat: Add mobile Android development docs

### PR DESCRIPTION
**Pull request by kylecarbs**
**Don't merge until 1.16.0 releases.** We need to support local -> remote port forwarding, and we don't in the latest released version.

### CODE REVIEW

File: environments/mobile-development.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+   ![Android SDK Install](../assets/android-sdk-missing.png)
+1. [Create and start a device.](https://developer.android.com/studio/run/managing-avds)
+   ![Android Device](../assets/android-avd.png)
+1. Set the installation path of your Android SDK to the environment variable `ANDROID_SDK_PATH`.
+   On MacOS, it's `~/Library/Android/sdk`.
+   ![Android SDK Path](../assets/android-sdk-path.png)
+1. Start the Android Debug Server on port 5555.
+   `$ANDROID_SDK_PATH/platform-tools/adb tcpip 5555`.
+1. Forward your Android Debug Server remotely.
+   `ssh -R 5555:127.0.0.1:5555 coder.<env name>`
```
Comment (cmoog) [inline environments/mobile-development.md:19]: first link to SSH setup docs

**Review by cmoog — CHANGES_REQUESTED** [review]

File: environments/mobile-development.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+   ```sh
+   brew install android-studio
+   ```
+
+1. Install the SDK.
+   ![Android SDK Install](../assets/android-sdk-missing.png)
+1. [Create and start a device.](https://developer.android.com/studio/run/managing-avds)
+   ![Android Device](../assets/android-avd.png)
+1. Set the installation path of your Android SDK to the environment variable `ANDROID_SDK_PATH`.
```
Comment (nathanpotter) [inline environments/mobile-development.md:23]: Would be good to show how to get to this preferences page to find this value, maybe something like:
Preferences > System Settings > Android SDK

Comment (nathanpotter) [inline environments/mobile-development.md:23]: This statement was a bit confusing as I went through it – are you telling me to set this config page option to `ANDROID_SDK_PATH` or are you saying go into your terminal and set an environment variable named `ANDROID_SDK_PATH` to the sdk path value that you found in this prefs page?

File: environments/mobile-development.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+   ```bash
+   $ adb devices
+   List of devices attached
+   emulator-5556
+   ```
+
+1. Build and run your Android applications remotely.
+
+   ```bash
+   ./gradlew android:installDebug
```
Comment (nathanpotter) [inline environments/mobile-development.md:59]: I didn't have any android code in my remote env, so this didn't do anything for me – may be good to show users how to transition their source code to being remote as well, potentially w/ coder sync

**Comment by bpmct** [comment]
I think we should add a "developer guides" section that supports stuff like this. I'm working on similar stuff that I'd love to put in docs.

**Comment by khorne3** [comment]
@kylecarbs Is this still going out with 1.16? If so, I can fix and merge it into the docs branch for the release

**Comment by kylecarbs** [comment]
That'd be awesome @khorne3 !

**Pull request merged**